tests/lib/devicetree/api/src/main.c

```diff
@@ -2133,6 +2133,20 @@ static void test_mbox(void)
 				  DT_NODELABEL(test_mbox_zero_cell)), "");
 }
 
+static void test_string_token(void)
+{
+#undef DT_DRV_COMPAT
+#define DT_DRV_COMPAT vnd_string_token
+	enum {
+		zero,
+		one,
+		two
+	};
+	zassert_equal(DT_INST_STRING_TOKEN_OR(0, val, zero), one, "");
+	zassert_equal(DT_INST_STRING_TOKEN_OR(1, val, one), zero, "");
+	zassert_equal(DT_INST_STRING_TOKEN_OR(2, val, two), two, "");
```

Comment:
It is invalid to make any assumptions about the instance numbers that are assigned to nodes. Please read https://docs.zephyrproject.org/latest/reference/devicetree/api.html#c.DT_INST for details, especially the documentation section beginning with "no other guarantees are made".

We also have independent test cases for each API macro in the test suite, so just having instance-number-based testing is not enough. Dedicated test cases for the non-INST variant are also required.

This also should be testing that the fallback value is not expanded if it is not needed -- search for BUILD_BUG_ON_EXPANSION in this same file for examples of how to test this.

Reply:
Done.
I increasing test scenarios and covering DT_INST_STRING_TOKEN_OR and DT_STRING_TOKEN_OR.
- Using DT_NODELABEL instead of instance number to test each existing node.
- Using DT_INST_FOREACH_STATUS_OKAY to loop through instances and check scenarios where default and non-default value is expanded.